### DISCUSSION TITLE
final destination priority

### DISCUSSION THREAD

**Discussion by hajes** [madMAx43v3r/chia-gigahorse#98] [Ideas]
As explained, in [discussion](https://github.com/madMAx43v3r/chia-gigahorse/discussions/91) `plotter` or `plot_sink` prioritizes a destination with more free space.

Scenario:

1. destination - one HDD 150-250MBs or 5-15min per plot
2. destination - stripped JBOD, RAID0 with 500-1000MBs

Slowly filling first destination if busy, use 2nd.

Current logic waits for both destination with same free space left. 2nd destination always finishes earlier than slow 1st destination.

Would be great to force write to preffered destination no matter how much space left (of course until full; use others)

**Comment by madMAx43v3r** [comment]
Actually the first priority is to use a disk that is not busy. So it should already work as you want it. The fast RAID disk will finish faster and thus get more writes.

**Reply by hajes** [reply]
Two scenarios that happened to me in the last 4 days of plotting.

1. 4x stripped JBOD 480-1000MBs sustainable write + 1 x HDD 120-250MBs

`plot_sink` fills 4xJBOD until both destinations have same free space left. I am not sure though, it looks like it filled always JBOD first because it is simple faster. Then it does zig-zag. Because 1x HDD will always be slower - end result is 4xJBOD finishes first, and plotter pause/copy to slow destination. I have forced to use slow disk....no idea though. My bet is, it simply fills faster, and then struggles with slower.

2. 3x stripped JBOD 360-800MBs + 1 x HDD.  Once 3x HDD JBOD gets almost full, sustainable speed drops below plotting speed, and `plot_sink` cycles destinations until plotter overflows `temp disk`, and pauses while waiting for destinations.


It is just a suggestion to fill slow HDDs as side kick :o) I usually do 4-8x JBODs so there is no issue with storing plots. But from time to time, there is just one HDD that will be sold soon, or replaced.

**Reply by madMAx43v3r** [reply]
it should fill all in parallel, of course the single HDD will be left at the end, there is nothing you can do about it...
